src/DynamoCore/Graph/Nodes/DummyNode.cs

```diff
@@ -227,6 +235,12 @@ private void SaveNode(XmlDocument xmlDoc, XmlElement nodeElement, SaveContext co
                     originalNode.AppendChild(nodeContent);
                     nodeElement.AppendChild(originalNode);
                 }
+                //if the node actually came from JSON we need to
+                //still preserve that incase this node was copied or undo/redone
+                else if (OriginalNodeContent is JObject)
+                {
+                    nodeElement.SetAttribute("OriginalNodeContent", OriginalNodeContent.ToString());
```

Comment:
I'm a bit confused here, so the XML save function will still be hit when we do undo/redo?

Reply:
yes